app/models/manageiq/providers/openshift/container_manager/refresh_parser.rb

```diff
@@ -180,20 +180,20 @@ def parse_openshift_image(openshift_image)
         new_result[:tag] ||= json["tag"] if json.keys.include?("tag")
 
         docker_metadata = openshift_image[:dockerImageMetadata]
+        docker_labels = parse_identifying_attributes(docker_metadata[:Config][:Labels], 'docker_labels', "openshift")
+        labels = parse_labels(openshift_image)
         new_result.merge!(
           :architecture          => docker_metadata[:Architecture],
           :author                => docker_metadata[:Author],
           :command               => docker_metadata[:Config][:Cmd],
           :entrypoint            => docker_metadata[:Config][:Entrypoint],
           :docker_version        => docker_metadata[:DockerVersion],
-          :exposed_ports         => parse_exposed_ports(
-            docker_metadata[:Config][:ExposedPorts]),
-          :environment_variables => parse_env_variables(
-            docker_metadata[:Config][:Env]),
+          :exposed_ports         => parse_exposed_ports(docker_metadata[:Config][:ExposedPorts]),
+          :environment_variables => parse_env_variables(docker_metadata[:Config][:Env]),
           :size                  => docker_metadata[:Size],
           :labels                => parse_labels(openshift_image),
-          :docker_labels         => parse_identifying_attributes(docker_metadata[:Config][:Labels],
-                                                                 'docker_labels', "openshift"),
+          :docker_labels         => docker_labels,
+          :tags                  => map_labels('ContainerImage', labels + docker_labels)
```

Comment:
Mitigating note: docker labels recommend `com.domain.reversed.label` format — if followed, collisions with k8s labels should be rare.

> I think the mapping code would map both.

Correction: UI [creates categories with single_value=true][1], so this either won't work, or will be "inconsistent" in DB.  Can fix, but let's understand behavior first.

> I feel doing this is a whole lot easier than adding label rate assignment code which is turning out as a lot but I may be wrong. Besides, tags are much more integrated into various miq features so this will solve multiple use case scenarios

Fair enough.  :+1: to smaller euwe changes and :+1: to less chargeback-special code.
Do we have a longer-term plan?

 - UX: will we show k8s labels / docker labels together as just "labels", or will we separate them?
 - Will we want independent k8s labels / docker labels mappings?
    I'm concerned that if we allow ambiguous "either one" mappings now, and want to separate later, a later migration won't be able to know which was meant.
    In any case, we can't do schema changes in euwe.
 - Will we eventually have direct MiqExpression (virtual column) support for docker labels?  Will they show merged with k8s labels or separate?
 - Will chargeback rate assignment eventually support [any kind of] labels?
   Is there any hope rate assigment will become MiqExpression based to stop playing catch-up?

[aside: Chargeback is not just technical & feature debt, it's a continuous attractor of new debt :-(.  IMHO we must allocate more time to gradually make it more based on the core infrastructure...]


[1]: https://github.com/ManageIQ/manageiq/blob/fe9e90af4434ebe18e8fe5e61fca2b0147647e16/app/controllers/ops_controller/settings/label_tag_mapping.rb#L147